framework/src/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1729,6 +1744,18 @@ public final String getOption(String name) {
         return getOption(name, null);
     }
 
+    /**
+     * Determines the boolean value of the option with the given name. Returns false if the option
+     * is not set.
+     *
+     * @param name the name of the option to check
+     * @see SourceChecker#getLintOption(String,boolean)
+     */
+    @Override
+    public final boolean getBooleanOption(String name) {
+        return Boolean.valueOf(getOption(name, "false"));
```

Comment:
`Boolean.valueOf` returns false if the input is null or a string other than `"true"`. It does not raise an exception.
Even if it did raise an exception, it would be nicer to raise a user-friendly error message that mentions the option name.
You do need to check for valid values here and in the other use of `Boolean.valueOf`.

Reply:
Thanks.  I had no idea that was the behavior.  (It seems wrong to me!)
I will update this to accommodate the actual behavior rather than what I thought it was.